testing/trino-tests/src/test/java/io/trino/execution/TestTableRedirection.java

```diff
@@ -0,0 +1,349 @@
+                .withListTables((session, schemaName) -> SCHEMA_TABLE_MAPPING.getOrDefault(schemaName, ImmutableSet.of()).stream()
+                        .map(name -> new SchemaTableName(schemaName, name))
+                        .collect(toImmutableList()))
+                .withStreamTableColumns((session, prefix) -> {
+                    List<TableColumnsMetadata> allColumnsMetadata = SCHEMA_TABLE_MAPPING.entrySet().stream()
+                            .flatMap(entry -> entry.getValue().stream().map(table -> new SchemaTableName(entry.getKey(), table)))
+                            .map(schemaTableName -> {
+                                if (REDIRECTIONS.containsKey(schemaTableName)) {
+                                    return new TableColumnsMetadata(schemaTableName, Optional.empty());
+                                }
+                                return new TableColumnsMetadata(schemaTableName, Optional.of(columnsGetter.apply(schemaTableName)));
+                            })
+                            .collect(toImmutableList());
+
+                    if (prefix.isEmpty()) {
+                        return allColumnsMetadata.stream();
+                    }
+
+                    String schema = prefix.getSchema().get();
+
+                    if (SCHEMAS.contains(schema)) {
+                        return allColumnsMetadata.stream()
+                                .filter(columnsMetadata -> columnsMetadata.getTable().getSchemaName().equals(schema))
+                                .filter(columnsMetadata -> prefix.getTable().map(columnsMetadata.getTable().getTableName()::equals).orElse(true));
+                    }
+
+                    return Stream.empty();
+                })
+                .withGetTableHandle((session, tableName) -> {
+                    if (SCHEMA_TABLE_MAPPING.getOrDefault(tableName.getSchemaName(), ImmutableSet.of()).contains(tableName.getTableName())
+                            && !REDIRECTIONS.containsKey(tableName)) {
+                        return new MockConnectorTableHandle(tableName);
+                    }
+                    return null;
+                })
+                .withGetViews(((connectorSession, prefix) -> ImmutableMap.of()))
+                .withGetColumns(schemaTableName -> {
+                    if (!REDIRECTIONS.containsKey(schemaTableName)) {
+                        return columnsGetter.apply(schemaTableName);
+                    }
+
+                    throw new RuntimeException("Columns do not exist for: " + schemaTableName);
+                })
+                .withRedirectTable(((connectorSession, schemaTableName) -> {
+                    return Optional.ofNullable(REDIRECTIONS.get(schemaTableName))
+                            .map(target -> new CatalogSchemaTableName(CATALOG_NAME, target));
+                }))
+                .build();
+    }
+
+    private static SchemaTableName name(String schemaName, String tableName)
```

Comment:
name -> schemaName

SchemaTableName class is probably more suitable place for this method